src/zombiereborn.cpp

```diff
@@ -242,32 +394,52 @@ void ZR_OnPlayerHurt(IGameEvent* pEvent)
 	int iDmgHealth = pEvent->GetInt("dmg_health");
 
 
-	//grenade and molotov knockbacks are handled by TakeDamage detours
+	// grenade and molotov knockbacks are handled by TakeDamage detours
 	if (!pAttackerController || !pVictimController || strcmp(szWeapon, "") == 0 || strcmp(szWeapon, "inferno") == 0 || strcmp(szWeapon, "hegrenade") == 0)
 		return;
 
 	if (pAttackerController->m_iTeamNum() == CS_TEAM_CT && pVictimController->m_iTeamNum() == CS_TEAM_T)
 		ZR_ApplyKnockback((CCSPlayerPawn*)pAttackerController->GetPawn(), (CCSPlayerPawn*)pVictimController->GetPawn(), iDmgHealth, szWeapon);
-
-	//if (pAttacker->m_iTeamNum == CS_TEAM_CT && pVictim->m_iTeamNum == CS_TEAM_T)
-	//	Message("lol");
-	//	//Knockback_Apply(pAttacker, pVictim, iDmgHealth, szWeapon);
-	//else if (szWeapon == "knife" && pAttacker->m_iTeamNum == CS_TEAM_T && pVictim->m_iTeamNum == CS_TEAM_CT)
-	//	Message("lol");
-	//	//Infect(pAttacker, pVictim, true, iHealth == 0);
 }
 
 void ZR_OnPlayerDeath(IGameEvent* pEvent)
 {
-	// To T TeamSwitch happening in ZR_OnPlayerSpawn
+	if (g_ZRRoundState == EZRRoundState::ROUND_START)
+		return;
+
+	CCSPlayerController *pVictimController = (CCSPlayerController*)pEvent->GetPlayerController("userid");
+	if (!pVictimController)
+		return;
+	CCSPlayerPawn *pVictimPawn = (CCSPlayerPawn*)pVictimController->GetPawn();
+	if (!pVictimPawn)
+		return;
+	
+	ZR_CheckWinConditions(pVictimPawn->m_iTeamNum() == CS_TEAM_CT);
+}
+
+void ZR_OnRoundFreezeEnd(IGameEvent* pEvent)
+{
+	ZR_StartInitialCountdown();
+}
 
-	/*ZEPlayer* pPlayer = g_playerManager->GetPlayerFromUserId(pEvent->GetInt("userid"));
+void ZR_CheckWinConditions(bool bCheckCT)
+{
+	int iTeamNum = bCheckCT ? CS_TEAM_CT : CS_TEAM_T;
 
-	if (pPlayer && !pPlayer->IsInfected())
+	// loop through each player, return early if both team has alive player
+	CCSPlayerPawn* pPawn = nullptr;
+	while (nullptr != (pPawn = (CCSPlayerPawn*)UTIL_FindEntityByClassname(pPawn, "player")))
 	{
-		CCSPlayerController* pController = (CCSPlayerController*)pEvent->GetPlayerController("userid");
-		pController->SwitchTeam(CS_TEAM_T);
-	}*/
+		if (!pPawn->IsAlive())
+			continue;
+		
+		if (pPawn->m_iTeamNum() == iTeamNum)
+			return;
+	}
+
+	// didn't return early, all players on one or both teams are dead.
+	// 8: CT win; 9: T wins; 10: draw
+	g_pGameRules->TerminateRound(5.0, bCheckCT ? 9u : 8u);
```

Comment:
Consider bringing something like [this in](https://github.com/alliedmodders/sourcemod/blob/master/plugins/include/cstrike.inc#L53-L77) rather than using magic values. I couldn't find any CS2 source for this, but seems to mostly be the same just +1.